monai/networks/nets/spade_network.py

```diff
@@ -297,12 +306,23 @@ def __init__(
         )
 
     def forward(self, seg, z: torch.Tensor | None = None):
-        if self.is_gan:
+        """
+        Args:
+            seg: input BxCxHxW[xD] semantic map on which the output is conditioned on
+            z: latent vector output by the encoder if self.is_vae is True. When is_vae is
+            False, z is a random noise vector.
+
+        Returns:
+
+        """
+        if not self.is_vae:
             x = F.interpolate(seg, size=tuple(self.latent_spatial_shape))
-            x = self.fc(x)
+            x = self.conv_init(x)
         else:
-            if z is None:
-                z = torch.randn(seg.size(0), self.opt.z_dim, dtype=torch.float32, device=seg.get_device())
+            if (
+                z is None and self.z_dim is not None
+            ):  # is_vae is Truee, but we can use the VAE-GAN as GAN in this function.
```

Comment:
```suggestion
            ):  # is_vae is True, but we can use the VAE-GAN as GAN in this function.
```

I didn't get the point of this line of comment.

Reply:
When flag is_vae is True, the network is a VAE-GAN, but it should be possible to sample from random noise as well, even when the network is a VAE-GAN (you either encode an input and then decode the latent resulting of that, or sample from noise). I've clarified this.